pkg/build/generator/generator.go

```diff
@@ -217,7 +217,8 @@ func (g *BuildGenerator) Instantiate(ctx kapi.Context, request *buildapi.BuildRe
 	if err != nil {
 		return nil, err
 	}
-
+	// update the build trigger information in the buildconfig
+	bc.Spec.Info = generateBuildTriggerInfo(request.BuildCauses)
```

Comment:
you should not be touching the buildconfig since this info is specific to the build instance, not the buildconfig.